packages/boot/src/mixins/boot.mixin.ts

```diff
@@ -71,6 +65,23 @@ export function BootMixin<T extends MixinTarget<Application>>(superClass: T) {
       );
     }
 
+    booted: boolean;
+
+    /**
+     * Override to detect and warn about starting without booting.
+     */
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    public async start(): Promise<void> {
+      await super.start();
+      if (!this.booted) {
+        process.emitWarning(
+          'App started without booting. Did you forget to call ' +
+            '`await app.boot()`?',
+        );
```

Comment:
Let's pass in `type` (such as 'LoopBackWarning') in `options` so that the printout will have `LoopBackWarning` prefix.